client/src/components/Overview/PrimaryImage.jsx

```diff
@@ -0,0 +1,161 @@
+      $('.carousel-control-next').addClass('hidden');
+    }
+    // cleanup func maybe an issue with switching product ID
+    return () => {
+      setActiveThumb(0);
+      setCurrentThumbs([0, 6]);
+      setCurrentPic(0);
+    };
+  }, [product]);
+
+  useEffect(() => {
+    setStylesPhotos(photoArr);
+  }, [style]);
+
+  const slideUp = () => {
+    if (currentThumbs[0] > 0) {
+      setCurrentThumbs([currentThumbs[0] - 7, currentThumbs[1] - 7]);
+    }
+  };
+
+  const slideDown = () => {
+    if (currentThumbs[1] < picCount) {
+      setCurrentThumbs([currentThumbs[0] + 7, currentThumbs[1] + 7]);
+    }
+  };
+
+  const makeActive = (e) => {
+    newInd = Number($(e.target).attr('data-index'));
+    $('.active-thumb').removeClass('active-thumb');
+    $(e.target).addClass('active-thumb');
+    setActiveThumb(newInd);
+    setCurrentPic(newInd);
+    $('.carousel').carousel(newInd);
+    $('.carousel-control-next').removeClass('hidden');
+    $('.carousel-control-prev').removeClass('hidden');
+    if (newInd === 0) {
+      $('.carousel-control-prev').addClass('hidden');
+    } else if (newInd === picCount - 1) {
+      $('.carousel-control-next').addClass('hidden');
+    }
+  };
+
+  const picUpdate = (button, totalPics, index) => {
+    if (button === 'next') {
+      if (index === totalPics - 1) {
+        $('.carousel-control-next').addClass('hidden');
+      }
+      $('.carousel-control-prev').removeClass('hidden');
+      if (index > currentThumbs[1]) {
+        slideDown();
+      }
+    }
+    if (button === 'prev') {
+      if (index === 0) {
+        $('.carousel-control-prev').addClass('hidden');
+      }
+      $('.carousel-control-next').removeClass('hidden');
+      if (index < currentThumbs[0]) {
+        slideUp();
+      }
+    }
+    setActiveThumb(index);
+    setCurrentPic(index);
+  };
+
+  const nextStyle = () => {
+    setStyle(style + 1);
+  };
+
+
+
+  const nextClick = () => {
+    $('.active-thumb').removeClass('active-thumb');
+    $('img').find(`[data-index='${currentPic + 1}']`).addClass('active-thumb');
+    picUpdate('next', picCount, Number(currentPic + 1));
+  };
+
+  const prevClick = () => {
+    $('.active-thumb').removeClass('active-thumb');
+    $('img').find(`[data-index='${currentPic - 1}']`).addClass('active-thumb');
+    picUpdate('prev', picCount, Number(currentPic - 1));
+  };
+
+  const imageExpand = () => {
+    console.log('in expand');
+    if (imageExpanded) {
+      $('.main-pic').removeClass('col-12').addClass('col-8');
+      setImageExpanded(false);
+      setTimeout(() => {
+        $('.main-details').removeClass('hidden').addClass('col-4');
+      }, 1000);
+    } else {
+      $('.main-pic').removeClass('col-8').addClass('col-12');
+      $('.main-details').removeClass('col-4').addClass('hidden');
+      setImageExpanded(true);
+    }
+  };
+
+
+
+  return (
+    <div id="main-image">
+      {/* <button onClick={ nextStyle }>nextStyle</button> */}
```

Comment:
If this is future functionality, I'd love to see a note that lets me know that, because, otherwise, it just looks like you forgot to take out a comment.

Reply:
I'm currently using this button to test for functionality with style changes (as the component that will do this has not yet been created). It will not be a part of deployment.